x/evm/keeper/statedb.go

```diff
@@ -181,6 +200,10 @@ func (k *Keeper) GetBalance(addr common.Address) *big.Int {
 // GetNonce retrieves the account with the given address and returns the tx
 // sequence (i.e nonce). The function performs a no-op if the account is not found.
 func (k *Keeper) GetNonce(addr common.Address) uint64 {
+	if k.HasStateError() {
+		return 0
+	}
+
 	ctx := k.Ctx()
 
 	cosmosAddr := sdk.AccAddress(addr.Bytes())
```

Comment:
set the error from GetSequence

Reply:
this part i am intentionally removed, since it really break some logic in the go-ethereum

https://github.com/ethereum/go-ethereum/blob/991384a7f6719e1125ca0be7fb27d0c4d1c5d2d3/core/vm/operations_acl.go#L66

which induce unwanted panic

Reply:
i guess we need to be careful to inject those error when we are doing so later on, what do you think about this?

Reply:
ok please add a relevant comment on the code describing why you left it out 👍

Reply:
👍 sure